src/library/scala/collection/GenMapLike.scala

```diff
@@ -119,17 +121,15 @@ trait GenMapLike[K, +V, +Repr] extends GenIterableLike[(K, V), Repr] with Equals
       (this eq that) ||
       (that canEqual this) &&
       (this.size == that.size) && {
-      try {
-        this forall {
-          case (k, v) => that.get(k.asInstanceOf[b]) match {
-            case Some(`v`) =>
-              true
-            case _ => false
-          }
+        val compare = new AbstractFunction1[(K, V), Boolean] with Function0[V] {
+          override def apply(v1: (K, V)): Boolean =
+            that.getOrElse(v1._1.asInstanceOf[b], this) == v1._2
```

Comment:
// Note: uncurry optimizes this to `get.getOrElse(..., this: Function0)`;  there is no extra lambda allocated.

Reply:
Done